pkg/apis/operator/v1alpha1/tektonconfig_lifecycle.go

```diff
@@ -21,107 +21,128 @@ import (
 	"knative.dev/pkg/apis"
 )
 
-var (
-	_ TektonComponentStatus = (*TektonConfigStatus)(nil)
+const (
+	ComponentsReady apis.ConditionType = "ComponentsReady"
+)
 
+var (
 	configCondSet = apis.NewLivingConditionSet(
-		DependenciesInstalled,
-		DeploymentsAvailable,
-		InstallSucceeded,
+		PreReconciler,
+		ComponentsReady,
+		PostReconciler,
 	)
 )
 
-// GroupVersionKind returns SchemeGroupVersion of a TektonConfig
-func (tp *TektonConfig) GroupVersionKind() schema.GroupVersionKind {
+func (tc *TektonConfig) GroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind(KindTektonConfig)
 }
 
-// GetCondition returns the current condition of a given condition type
-func (tps *TektonConfigStatus) GetCondition(t apis.ConditionType) *apis.Condition {
-	return configCondSet.Manage(tps).GetCondition(t)
+func (tc *TektonConfig) GetGroupVersionKind() schema.GroupVersionKind {
+	return SchemeGroupVersion.WithKind(KindTektonConfig)
```

Comment:
Any reason to have both ?

Reply:
`GroupVersionKind` can be removed  now.. will update !

Reply:
in generated code and internally in knative/pkg `` is used 
`GetGroupVersionKind` is defined so that we can do
```
tektonTriggerinformer.Get(ctx).Informer().AddEventHandler(cache.FilteringResourceEventHandler{
			FilterFunc: controller.FilterController(&v1alpha1.TektonConfig{}),
			Handler:    controller.HandleAll(impl.EnqueueControllerOf),
		})
```
so lets keep `GroupVersionKind` also 
i will take a deep look and then remove